### PR TITLE
fix(TextLink): Make touchable, add ‘inline’, remove ‘interaction’ text

### PR DESCRIPTION
**Pull request by markdalgleish**
`TextLink` is now touchable by default and renders a `Text` component internally on your behalf.

For example:

```js
<Card>
  <TextLink href="...">This is a touchable link.</TextLink>
</Card>
```

Note, the `interaction` text size has now been deprecated.

## Migration Guide

To maintain existing inline link behaviour, you will need to use the `inline` prop:

```diff
<Text>
-  The last word of this sentence is an <TextLink>inline link</TextLink>
+  The last word of this sentence is an <TextLink inline>inline link</TextLink>
</Text>
```

### CODE REVIEW

File: lib/components/private/getCheckboxRadioSize.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export default (theme: Theme) => {
   // We currently don't support responsive checkboxes and
   // radio buttons, but nobody actually needs it (so far)
   const scale = theme.tokens.text.standard.mobile.size / 28;
```
Comment (michaeltaranto) [inline lib/components/private/getCheckboxRadioSize.ts]: 🤔

File: lib/themes/theme.d.ts [hunk 2]
```diff
@@ -171,6 +171,7 @@ type SpacingVariants =
   | 'xlarge'
   | 'xxlarge';
 type HorizontalSpacingVariants = SpacingVariants | 'gutter';
+type VerticalPaddingVariants = SpacingVariants | 'standardTouchableText';
```
Comment (michaeltaranto) [inline lib/themes/theme.d.ts]: Didn't we decide on these types being singular?

File: lib/components/private/getCheckboxRadioSize.ts (hunk 1 above)
Comment (markdalgleish) [reply lib/components/private/getCheckboxRadioSize.ts]: 🤔

File: lib/themes/theme.d.ts (hunk 2 above)
Comment (markdalgleish) [reply lib/themes/theme.d.ts]: Fixed. Also updated the other two spacing types to be consistent.

**Pull request merged**